src/main/java/manageezpz/logic/commands/EditTaskCommand.java

```diff
@@ -108,30 +112,78 @@ public CommandResult execute(Model model) throws CommandException {
                 // Should not reach this as there are only three types of tasks
                 throw new CommandException(MESSAGE_INVALID_TASK_TYPE);
             }
-
             model.setTask(currentTask, updatedTask);
             return new CommandResult(String.format(MESSAGE_EDIT_TASK_SUCCESS, updatedTask));
         } catch (ParseException pe) {
             throw new CommandException(pe.getMessage() + "\n\n" + EditTaskCommand.MESSAGE_USAGE, pe);
         } catch (DuplicateTaskException de) {
-            throw new CommandException(String.format(MESSAGE_DUPLICATE_TASK, this.desc) + MESSAGE_USAGE);
+            throw new CommandException(String.format(MESSAGE_DUPLICATE_TASK, this.desc) + MESSAGE_USAGE, de);
+        }
+    }
+
+    /**
+     * If the value after a prefix is empty, return false.
+     * Else, return true.
+     */
+
+    private boolean ensureFormatCompliance(HashMap<String, Boolean> prefixStatusHash,
+                                   String desc, String date, String time) {
+        boolean isFormatOkay = true;
+        HashMap<String, String> inputStatusHash = new HashMap<>();
+        inputStatusHash.put("description", desc);
+        inputStatusHash.put("date", date);
+        inputStatusHash.put("datetime", time);
+
+        String[] statusArr = {"description", "date", "datetime"};
+
+        for (String s : statusArr) {
+            boolean status = prefixStatusHash.get(s);
+            String input = inputStatusHash.get(s).trim();
+            if (status && input.isEmpty()) {
+                isFormatOkay = false;
+                break;
+            }
         }
+        return isFormatOkay;
     }
 
+    /**
+     * Updates a Todo task.
+     *
+     * @param currentTask Current Todo task that is to be updated
+     * @param desc New description of the Todo Task
+     * @return Updated Todo task
+     */
+
     private Task updateTodo(Todo currentTask, String desc) throws ParseException {
         Todo updatedToDoTask = new Todo(currentTask);
-
-        if (!desc.isEmpty()) {
-            Description newDesc = ParserUtil.parseDescription(desc);
-            updatedToDoTask.setDescription(newDesc);
+        if (prefixStatusHash.get("date") || prefixStatusHash.get("datetime")) {
+            throw new ParseException(MESSAGE_EDIT_TODO_TASK_NO_DATE_AND_TIME_VALUES);
```

Comment:
Change this to `throw new ParseException(String.format(MESSAGE_EDIT_TODO_TASK_NO_DATE_AND_TIME_VALUES, MESSAGE_USAGE));`

This is so that after this error message, we will show them the MESSAGE_USAGE again.